DMCompiler/Compiler/DMPreprocessor/DMPreprocessor.cs

```diff
@@ -135,27 +141,46 @@ public sealed class DMPreprocessor : IEnumerable<Token> {
                         if (!_lastIfEvaluations.TryPop(out var _))
                             DMCompiler.Emit(WarningCode.BadDirective, token.Location, "Unexpected #endif");
                         break;
-                    case TokenType.DM_Preproc_Identifier:
-                        _currentLineWhitespaceOnly = false;
-                        if(!TryMacro(token)) {
+                    case TokenType.DM_Preproc_Identifier: {
+                        if (TryMacro(token)) {
+                            break;
+                        }
+
+                        // Otherwise treat it like any other normal token
+                        goto case TokenType.DM_Preproc_Number;
+                    }
+                    case TokenType.DM_Preproc_Punctuator:
+                        if(token.Text == ";") {
```

Comment:
Make this its own punctuator token.

Once you do that you can also merge this with the `case TokenType.DM_Preproc_Punctuator_RightParenthesis` below by replacing the `_canUseDirective = false;` with `_canUseDirective = (token.Type == TokenType.DM_Preproc_Punctuator_Semicolon);`.